### PR TITLE
Refactor getExpiryForCreation call to a common place

### PR DESCRIPTION
**Pull request by henri-tremblay**

### CODE REVIEW

File: core/src/main/java/org/ehcache/core/config/ExpiryUtils.java [hunk 1]
```diff
@@ -173,5 +177,39 @@ public static long getExpirationMillis(long now, Duration duration) {
...
+   * @return the duration returned by to pass to {@link ExpiryPolicy#getExpiryForCreation(Object, Object)}, {@code null}
+   * if the call throws an exception, if the returned duration is {@code null} or if it is lower or equal to 0
+   */
+  public static <K, V> Duration getExpiryForCreation(K key, V value, ExpiryPolicy<? super K, ? super V> expiry) {
+    Duration duration;
+    try {
+      duration = expiry.getExpiryForCreation(key, value);
+    } catch (RuntimeException e) {
+      LOG.error("Expiry computation caused an exception - Expiry duration will be 0", e);
+      return null;
```
[the remark this answers is not in the document]
Comment (henri-tremblay) [reply core/src/main/java/org/ehcache/core/config/ExpiryUtils.java]: Ok. Will be a tiny bit slower though

[the remark this answers is not in the document]
Comment (henri-tremblay) [reply core/src/main/java/org/ehcache/core/config/ExpiryUtils.java]: But I agree, it is much more meaningful

File: core/src/main/java/org/ehcache/core/config/ExpiryUtils.java [hunk 2]
```diff
@@ -173,5 +177,39 @@ public static long getExpirationMillis(long now, Duration duration) {
...
+    } catch (RuntimeException e) {
+      LOG.error("Expiry computation caused an exception - Expiry duration will be 0", e);
+      return Duration.ZERO;
+    }
+
+    try {
+      if (Duration.ZERO.compareTo(duration) >= 0) {
+        return Duration.ZERO;
+      }
+    } catch(NullPointerException e) {
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/config/ExpiryUtils.java]: I think I'd prefer an explicit null-check here.

Comment (henri-tremblay) [reply core/src/main/java/org/ehcache/core/config/ExpiryUtils.java]: :-) I thought about it and decided that it is unlikely to be null so this is fine. But as you wish

**Comment by chrisdennis** [comment]
If you can rebase/fix this I'll merge it.

**Comment by henri-tremblay** [comment]
rebased

**Comment by henri-tremblay** [comment]
test this please

**Pull request merged**